src/components/common/UserProfileEdit/AdvancedSettingsRow.tsx

```diff
@@ -0,0 +1,190 @@
+
+import stylesAdvance from './UserAdvancedSettings.css';
+
+const displayName = 'common.UserProfileEdit.AdvancedSettingsRow';
+
+const MSG = defineMessages({
+  metaDescription: {
+    id: `${displayName}.metaDescription`,
+    defaultMessage: `To connect directly to Gnosis chain and pay for your own transactions, disable this option.`,
+  },
+  metaDescGlobalOff: {
+    id: `${displayName}.metaDescGlobalOff`,
+    defaultMessage: `Metatransactions are disabled globally.`,
+  },
+  labelMetaTx: {
+    id: `${displayName}.labelMetaTx`,
+    defaultMessage: `Metatransactions ({isOn, select,
+      true {active}
+      other {inactive}
+    })`,
+  },
+  metaTooltip: {
+    id: `${displayName}.metaTooltip`,
+    defaultMessage: `Metatransactions are turned on by default.
+    If you would rather connect directly to the chain,
+    and pay for your own transactions, you can turn them off
+    by switching the toggle at any time. {br}{br} Please note,
+    this setting is stored locally in your browser,
+    if you clear your cache you will need to turn Metatransactions off again.`,
+  },
+  customEndpoints: {
+    id: `${displayName}.customEndpoints`,
+    defaultMessage: `Enable custom endpoints ({isOn, select,
+      true {active}
+      other {inactive}
+    })`,
+  },
+  endpointsDescription: {
+    id: `${displayName}.metaDescription`,
+    defaultMessage: `If you prefer maximum decentralisation, you may use your own custom endpoints for Colony.`,
+  },
+  labelRPC: {
+    id: `${displayName}.labelRPC`,
+    defaultMessage: 'Gnosis Chain RPC',
+  },
+  RPCTooltip: {
+    id: `${displayName}.RPCTooltip`,
+    defaultMessage: `You will be able to toggle the Gnosis Chain RPC on once you have successfully validated that the endpoint works.`,
+  },
+  validate: {
+    id: `${displayName}.validate`,
+    defaultMessage: 'Validate',
+  },
+});
+
+interface MetatransactionsDisabledProps {
+  metatransactionsDisabled: boolean;
+}
+
+const MetatransactionsDisabled = ({
+  metatransactionsDisabled,
+}: MetatransactionsDisabledProps) =>
+  metatransactionsDisabled ? (
+    <div className={stylesAdvance.metaDesc}>
+      <FormattedMessage {...MSG.metaDescGlobalOff} />
+    </div>
+  ) : null;
+
+const CustomEndpointInput = () => {
+  const {
+    watch,
+    formState: { isValid },
+  } = useFormContext();
+  const decentralizedMode: boolean = watch(SlotKey.DecentralizedMode);
+  const customRpc: string = watch(SlotKey.CustomRPC);
+
+  return (
+    <>
+      <Input
+        label={MSG.labelRPC}
+        appearance={{ colorSchema: 'grey' }}
+        name={SlotKey.CustomRPC}
+        disabled={!decentralizedMode}
+        formattingOptions={noSpaces}
+        value={customRpc}
+      />
+      <div className={stylesAdvance.validateButtonContainer}>
+        <Button
+          text={MSG.validate}
+          disabled={!decentralizedMode || !isValid}
+          // @TODO: validation to be added later
+        />
+      </div>
+    </>
+  );
+};
```

Comment:
I think this needs to be generic.

While currently we only a single input on the _Advanced Tab_, the plan is to have multiple _(RPC endpoint, subgraph, IPFS, block ingestor, maybe even a custom db, etc)_